openssl/src/ssl/mod.rs

```diff
@@ -870,6 +870,23 @@ impl SslContextBuilder {
         }
     }
 
+    /// Add the provided CA certificate to the list sent by the server to the client when
+    /// requesting client-side TLS authentication.
+    ///
+    /// This corresponds to [`SSL_CTX_add_client_CA`].
+    ///
+    /// [`SSL_CTX_add_client_CA`]: https://www.openssl.org/docs/man1.0.2/man3/SSL_CTX_set_client_CA_list.html
+    #[cfg(not(libressl))]
+    pub fn add_client_ca(&mut self, cacert: &mut X509) -> Result<(), ErrorStack> {
```

Comment:
This should take a `&X509Ref`, not a `&mut X509`.

Reply:
I agree about the `X509Ref`, but I disagree about removing the `mut`.  The OpenSSL API does not mark that API as `const` so for safety I have to take the CA as `mut` (so it can become a `*mut X509` in the `ffi`).

Reply:
OpenSSL's APIs are extremely not const-correct, and in any case, C's notion of const is not a 1:1 match to Rust's notion of const.

Reply:
Agreed, but, as the API doesn't say `const` I can't guarantee that it won't modify the certificate... and rust/llvm/gep get very unhappy if you alias mutability.

Reply:
This entire library ignores the const annotations in the OpenSSL API in favor of actually inspecting what the implementation does, and what it's documented to do. There is nothing unique about this one method.

Reply:
Providing those guarantees is exactly the job of the author of safe bindings over a C APi.

Reply:
Coolio, I'll make the change to `&` then